Switch to GHA CI workflow

internalsystemerror opened the pull request:
Switch from Travis CI to use GitHub Actions according to: https://gist.github.com/weierophinney/b003e50c3c2667d08076caf31ebd36a4.

This closes #41

Commit:
Switch to GHA CI workflow

Signed-off-by: Gary Lockett <gary@creativecow.uk>

Commit:
Add array type configuration

Signed-off-by: Gary Lockett <gary@creativecow.uk>

Commit:
Correct build status image path

Signed-off-by: Gary Lockett <gary@creativecow.uk>

internalsystemerror commented:
I'm getting different results locally compared to here ... ~~I actually pass CS checks locally since `phpcs.xml` is used whereas here it seems to be using `phpcs.xml.dist` instead (just a hypothesis)~~ (edit: fixed). And I'm also failing a lot more tests locally, feel free to take over this PR as needed otherwise I'll check back on this later.

brayniverse commented on test/MezzioInstallerTest/HomePageResponseTest.php:
```diff
@@ -218,7 +229,7 @@ public function installCasesProvider(): Generator
                 }
 
-                foreach ($this->intallTypes as $intallType) {
+                foreach ($this->installTypes as $intallType) {
```
You could correct the `$intallType` typos here while you're fixing `$this->intallType`.

```suggestion
                foreach ($this->installTypes as $installType) {
```

brayniverse commented on test/MezzioInstallerTest/HomePageResponseTest.php:
```diff
@@ -307,7 +318,7 @@ public function rendererlessInstallCasesProvider(): Generator
                 $routerDocs   = $this->expectedRouterAttributes[$routerClass]['routerDocs'];
 
-                foreach ($this->intallTypes as $intallType) {
+                foreach ($this->installTypes as $intallType) {
```
```suggestion
                foreach ($this->installTypes as $installType) {
```

The typo is here as well.

Commit:
correct typo in tests

Signed-off-by: Gary Lockett <gary@creativecow.uk>

internalsystemerror commented:
I believe this is now ready for review

weierophinney committed:
qa: exclude locked run on PHP 7.3

Due to chubbyphp/container, as lockfile uses v2, which requires 7.4+.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
qa: fix CS errors

- Removes errantly provided phpcs.xml, which duplicated phpcs.xml.dist, but poorly.
- Fixes errors discovered during phpcs run.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
qa: exclude generated/manipualated files from CS checks

- config file gets updated as we are not using `--no-scripts` during composer install
- routes file is copied from source during composer install

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>
